pages/profile/[username].tsx

```diff
@@ -0,0 +1,165 @@
+import { BaseContainer } from "@components/Page/BaseContainer";
+import { Helmet } from "react-helmet";
+import React, { useEffect, useRef, useState } from "react";
+import LayoutContainer from "@components/LayoutContainer";
+import Image from "next/image";
+import { Header } from "@components/Header";
+import Toggle from 'react-toggle'
+import { useRouter } from "next/router";
+import {  MessageEnum } from "@shared/enum";
+import { GetServerSidePropsContext } from "next";
+import { toast } from "react-toastify";
+import { useGetProfile } from "@services/profile/profileHooks";
+import { LoaderWrapper } from "@components/LoaderWrapper";
+import useToastHook from "@hooks/toast/useToastHook"
+import getConfig from "next/config";
+
+interface PageProps {
+    username?: string
+}
+
+const { publicRuntimeConfig } = getConfig()
+
+export default function Profile(props: PageProps) {
+    const { data, isLoading } = useGetProfile(props.username);
+    const [message, setMessage] = useState('')
+    const router = useRouter();
+    useEffect(() => {
+        if (data) {
+            localStorage.setItem(MessageEnum.targetUser, data.user_id);
+        }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Storing sensitive user data like `user_id` in local storage is not a secure practice. Local storage is accessible by any script on your page, which opens up the possibility of it being accessed maliciously. Consider using HTTP-only cookies or storing such information server-side and providing only necessary tokens to the client side.

```diff
- localStorage.setItem(MessageEnum.targetUser, data.user_id);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->